cray/modules/bos/cli.py

```diff
@@ -50,31 +73,104 @@
 # Add --file parameter for specifying session template data
 def create_templates_shim(func):
     """ Callback function to custom create our own payload """
-    def _decorator(file, name, **kwargs):
+    def _decorator(file, **kwargs):
         if not file.get('value'):
-            return func(name=name, **kwargs)
+            return func(**kwargs)
         with open(file["value"], 'r', encoding='utf-8') as f:
             data = json.loads(f.read())
         payload = data
-        if name:
-            payload['name'] = name['value']
+        if 'name' in kwargs:
+            payload['name'] = kwargs['name']['value']
+            del kwargs['name']
         # Hack to tell the CLI we are passing our own payload; don't generate
         kwargs[FROM_FILE_TAG] = {"value": payload, "name": FROM_FILE_TAG}
         return func(**kwargs)
     return _decorator
 
 
-def setup_template_from_file(bos_cli):
+def setup_template_from_file(command):
     """ Adds a --file parameter for session template creation """
-    command = bos_cli.commands['sessiontemplate'].commands['create']
     option('--file', callback=_opt_callback, type=str, default='', metavar='TEXT',
            help="A file containing the json for a template")(command)
     params = [command.params[-1]]
     for param in command.params[:-1]:
-        if not param.help or 'DEPRECATED' not in param.help:
+        if not getattr(param, 'help', None) or 'DEPRECATED' not in param.help:
             params.append(param)
     command.params = params
     command.callback = create_templates_shim(command.callback)
 
 
-setup_template_from_file(cli)
+def updatemany_data_handler(args):
+    """ Handler to override the api action taken for updatemany """
+    _, path, data = args
+    return "PATCH", path, data
+
+
+def create_patch_shim(func):
+    """ Callback function to custom create our own payload """
+    def _decorator(filter_ids, filter_session, patch, enabled, **kwargs):
+        filter_ids=filter_ids["value"]
+        filter_session=filter_session["value"]
+        if not (filter_ids or filter_session):
+            raise Exception('Either the ids or session filter must be provided')
+        if filter_ids and filter_session:
+            raise Exception('Only one of the two filter options can be provided')
+        payload = {}
+        if filter_ids:
+            payload["filters"] = {"ids": filter_ids}
+        if filter_session:
+            payload["filters"] = {"session": filter_session}
+        if patch["value"]:
```

Comment:
Is the "value" key guaranteed to be here? Otherwise, we'll get a KeyError.

Reply:
It's guaranteed.  patch is required and will be a dictionary with the keys "key" and "value".  Not sure why it's done this way, but it is.